test/framework/run.py

```diff
@@ -118,6 +118,22 @@ def test_run_cmd_simple(self):
         self.assertEqual(True, run_cmd("echo hello", simple=True))
         self.assertEqual(False, run_cmd("exit 1", simple=True, log_all=False, log_ok=False))
 
+    def test_run_cmd_cache(self):
```

Comment:
What about a test for the new `inp` argument to `run_cmd`?

Reply:
good point, added in #bc4be4154